pkg/workflow_template.go

```diff
@@ -608,7 +612,7 @@ func (c *Client) ListWorkflowTemplates(namespace string, paginator *pagination.P
 }
 
 func (c *Client) ArchiveWorkflowTemplate(namespace, uid string) (archived bool, err error) {
-	workflowTemplate, err := c.getWorkflowTemplate(namespace, uid, 0)
+	workflowTemplate, err := c.getWorkflowTemplate(namespace, uid, 0) //version=0 means latest
```

Comment:
Maybe we should add a `getLatestWorkflowTemplate` method so we don't need to pass in a weird 0 for version. We'd still have that method, but add this new convenience one. 

What do you guys think?
@aleksandrmelnikov @rushtehrani

Reply:
Generally, I like the idea. Have to use go docs to tell the user to use it ;).

Reply:
Refactored